docs/changelog2.rst

```diff
@@ -0,0 +1,267 @@
+* Optionally output to a user-specified array (``out`` keyword argument) when constructing
+  geometries from ``indices`` (#380).
+* Enable bulk construction of geometries with different number of coordinates
+  by optionally taking index arrays in all creation functions (#230, #322, #326, #346).
+
+
+API changes (deprecated in 1.8)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+See :ref:`migration` for more details on how to update your code.
+
+Recommended to first upgrade to Shapely 1.8 and resolve all deprecation
+warnings before upgrading to Shapely 2.0.
+
+Summary of changes:
+
+* Geometries are now immutable and hashable.
+* Multi-part geometries no longer behave as "sequences". This means that they
+  no longer have a length, are not iterable and indexable anymore. Use ``.geoms``
+  attribute instead to access the parts.
+* Geometries no longer directly implement the numpy array interface. To
+  convert to an array of coordinates, use `.coords` instead
+  (`np.asarray(geom.coords)`).
+* Consistent creation of empty geometries (for exampe ``Polygon()`` now
+  actually creates an empty Polygon instead of an empty geometry collection).
+* The following attributes and methods on the Geometry classes were
+  deprecated and are now removed:
+  * ``array_interface()`` and ``ctypes``
+  * ``asShape()``, and the adapters classes to create geometry-like proxy
+    objects (use ``shape()`` instead).
+  * ``empty()`` method
+
+Breaking API changes
+^^^^^^^^^^^^^^^^^^^^
+
+Some additional backwards incompatible API changes were included in Shapely
+2.0 that were not yet deprecated in Shapely 1.8:
+
+* The default of the ``preserve_topology`` keyword of ``simplify()`` changed
+  to True (#1392).
+* A ``GeometryCollection`` that consists of all empty sub-geometries now
+  returns those empty geometries in ``.geoms`` (instead of returning an empty
+  list) (#1420).
+* The unused ``shape_factory()`` method and ``HeterogeneousGeometrySequence``
+  class are removed (#1421).
+* The undocumted ``__geom__`` attribute is removed. To access the raw GEOS pointer,
+  the ``_geom`` attribute is still present (1417).
+
+New features
+^^^^^^^^^^^^
+
+More informative repr with (truncated) WKT
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+Support for fixed precision model for geometries and in overlay functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* `set_precision` to conform a geometry to a certain grid size, and this will then also be used by subsequent overlay methods
+* `grid_size` keyword in overlay methods (`intersection`, `union`, `difference`, etc)
+
+
+* Addition of ``get_precision`` to get precision of a geometry and ``set_precision``
+  to set the precision of a geometry (may round and reduce coordinates) (#257)
+* Addition of ``grid_size`` parameter to specify fixed-precision grid for ``difference``,
+  ``intersection``, ``symmetric_difference``, ``union``, and ``union_all`` operations for
```

Comment:
This looks like a very interesting new feature. Are there any documents the team has floating around that discuss in more detail? Seems to move closer to fixed number type operations which might offer big improvements in robustness. However, with a bit of experience using CGAL usage of exact number types you often pay a hefty price in performance.

Looking around, I can see the GEOS 3.9.0 introduced this using the [OverlayNG](http://lin-ear-th-inking.blogspot.com/2021/02/overlayng-and-invalid-geometry.html). We do lots of intersections with lots of "real world" geometries, so this could be big for us.

Reply:
I am not aware of something on the Shapely side, but Martin Davis has some blogposts about his work (this is indeed enabled by the OverlayNB work). The main one: http://lin-ear-th-inking.blogspot.com/2020/05/jts-overlay-next-generation.html (and some others as well: http://lin-ear-th-inking.blogspot.com/search/label/overlay)

The JTS docs also have some more detail as the GEOS one: https://locationtech.github.io/jts/javadoc/index.html

Reply:
(and it would certainly be good to provide more details in the shapely docs as well, as this improvement is indeed a big thing)

Reply:
Played around a little bit extending this [geopandas PR](https://github.com/geopandas/geopandas/pull/2184) to stress test the `set_precision`. Seems to have some very nice properties when you need to snap to integer coordinates (e.g. for vector tiles). `1.8.2` produces lots of invalid geometries with a naive snap function, whereas `2.0a1` and `set_precision(geom, 1)` always returns valid.

See https://github.com/bretttully/shapely-v2

Reply:
Example of some stressful inputs to overlay functions

![image](https://user-images.githubusercontent.com/5830269/183032148-4f7f7401-80c3-446a-8dfb-c588a64cf0f9.png)
![image](https://user-images.githubusercontent.com/5830269/183032184-3f974808-8d90-4b5f-bd08-452dda9ce97e.png)
![image](https://user-images.githubusercontent.com/5830269/183032170-929e767b-f0c7-4d7b-9b0b-17dce5b64fb2.png)